### ISSUE TITLE
Allow defining pyproject.toml argument check-filenames as a boolean

### ISSUE THREAD

**Issue by amarvin** [codespell-project/codespell#3337]
Currently, I can define my `pyproject.toml` with `check-filenames` as a boolean argument:

```toml
[tool.codespell]
check-filenames = true
```

and that works fine when invoking `codespell` from command line, but fails as a `precommit` hook. If my `.pre-commit-config.yaml` is

```yaml
repos:
-   repo: https://github.com/codespell-project/codespell
    rev: v2.2.4
    hooks:
    -   id: codespell
        additional_dependencies:
        -   tomli
```

then `precommit` during a Git commit gives an error:

```
codespell................................................................Failed
- hook id: codespell
- exit code: 2

usage: codespell.EXE [-h] [--version] [-d] [-c] [-w] [-D DICTIONARY]
                     [--builtin BUILTIN-LIST] [--ignore-regex IGNORE_REGEX]
                     [-I FILE] [-L WORDS] [--uri-ignore-words-list WORDS]
                     [-r REGEX] [--uri-regex URI_REGEX] [-s] [--count]
                     [-S SKIP] [-x FILE] [-i INTERACTIVE] [-q QUIET_LEVEL]
                     [-e] [-f] [-H] [-A LINES] [-B LINES] [-C LINES]
                     [--config CONFIG] [--toml TOML]
                     [files ...]
codespell.EXE: error: unrecognized arguments: True
```

but it works fine if I change `pyproject.toml` to

```toml
[tool.codespell]
check-filenames = ''
```

then the `precommit` hook passes:

```
codespell................................................................Passed
```

Relates to https://github.com/codespell-project/codespell/pull/3307#issuecomment-1914884352, where the similar boolean argument `count` is allowed to be a boolean in `pyproject.toml`. It'd be great to have consistent behavior for all boolean arguments.

**Comment by DimitriPapadopoulos** [comment]
I think #3046 solved this issue. It's just that this change has not been included in a _codespell_ release, not to mention _pre-commit_.

Does it make sense?

**Comment by amarvin** [comment]
Awesome! Yes, I think you're right @DimitriPapadopoulos that this should be resolved by the next release due to #3046.

**Comment by amarvin** [comment]
Duplicate of #2839
